bowtie/_report.py

```diff
@@ -1,6 +1,8 @@
 from __future__ import annotations
 
 from datetime import UTC, datetime
+from functools import cmp_to_key
```

Comment:
This function should essentially never be used (it's there to support legacy `cmp` functions).
The version comparison below looks super complicated compared to simply splitting and trying to call int. I'll try to have a look in a bit and give you tips, but I'd expect this to be ~4 lines long essentially.